### PR TITLE
Block submissions to postgraduate courses when candidate has no degrees

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

As part of enabling TDA applications, we have made a change in the ‘Your Details’ in apply.

In the ‘Degree’ section, we have added a radio button question to the initial screen of the section. [See example](https://trello.com/1/cards/66cf30bbfb30dfdad251bc7c/attachments/66cf3243418f19737d45128d/download/Screenshot_2024-08-28_at_15.18.49.png). This allows the section to be marked as ‘Complete’ if a candidate does not have a degree.

We need now to make sure we validate on submission that candidates should have degrees when submitting to a postgraduate course.

## Changes proposed in this pull request

Screenshot 2024-09-06 at 17 40 06

## Guidance to review

To test this feature locally or on the review app you need:

1. Set the cycle switcher (visiting /support/settings/cycles) and choose any of these options: "Find has reopened" or "Apply has reopened"
2. Activate the feature flag "Teacher degree apprenticeship"
3. Choose option no on the degrees section
4. Try to submit a postgraduate course.

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[TDA: Block submissions to postgraduate courses when a candidate has marked they do not have a degree](https://trello.com/c/Xi9Ut0dJ/2243-tda-block-submissions-to-postgraduate-courses-when-a-candidate-has-marked-they-do-not-have-a-degree)

File: app/validators/incomplete_postgraduate_course_details_validator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class IncompletePostgraduateCourseDetailsValidator < ActiveModel::EachValidator
+  include ActionView::Helpers::UrlHelper
+  include GovukLinkHelper
+  include GovukVisuallyHiddenHelper
+
+  def validate_each(record, attribute, application_choice)
+    return unless application_choice.application_form.teacher_degree_apprenticeship_feature_active?
+    return if application_choice.course.undergraduate?
```
Comment (tomas-stefano) [inline app/validators/incomplete_postgraduate_course_details_validator.rb:8]: I put those two conditions in different lines to become easier to read and understand very fast. 
Let me know if you prefer everything in one liner.

**Comment by CatalinVoineag** [comment]
Dev tested it on Review.

It seems like you can apply for a TDA course with Postgraduate degree, if you apply for it before you completed your degree details. Is this what we want?


https://github.com/user-attachments/assets/f253fb59-d584-45a2-a331-14efd454084b

**Comment by tomas-stefano** [comment]
> Dev tested it on Review.
> 
> It seems like you can apply for a TDA course with Postgraduate degree, if you apply for it before you completed your degree details. Is this what we want?
> 
>  Screencast.2024-09-09.13.04.02.mp4

Yes, that's expected. 

In this PR, the scenario to be tested is:

1. Answer "No, I do not have a degree"
2. Try to submit a postgraduate (non TDA) course
3. It shouldn't let you submit.
4. Return to the degree "Yes, I do have a degree"
5. Add your degree
6. Complete the section
7.  Try to submit a postgraduate (non TDA) course - It should be succcessful.

**Review by CatalinVoineag — APPROVED** [review]
Dev tested it on review, works as expected 👍🏻

**Pull request merged**